e_chil.c

```diff
@@ -950,30 +950,26 @@ static EVP_PKEY *hwcrhk_load_privkey(ENGINE *eng, const char *key_id,
     bn_e = hwcrhk_mpi_mpi2bn(e, NULL);
     bn_n = hwcrhk_mpi_mpi2bn(n, NULL);
 
-    hwcrhk_mpi_free(e);
-    hwcrhk_mpi_free(n);
```

Comment:
Actually, instead of moving these, add these lines:

```
    e = NULL;
    n = NULL;
```

The norm being that for freeing functions, NULL is a perfectly valid (no-op) value.  I see that `hwcrhk_mpi_free` has a NULL check already.

Reply:
With that, it's not necessary to rearrange the code so much...  the added freeing functions in error cases should suffice.